pkg/auth/handler/webapp/setup_magic_link_otp.go

```diff
@@ -0,0 +1,112 @@
+package webapp
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/authgear/authgear-server/pkg/auth/handler/webapp/viewmodels"
+	"github.com/authgear/authgear-server/pkg/auth/webapp"
+	"github.com/authgear/authgear-server/pkg/lib/interaction"
+	"github.com/authgear/authgear-server/pkg/util/httproute"
+	"github.com/authgear/authgear-server/pkg/util/template"
+	"github.com/authgear/authgear-server/pkg/util/validation"
+)
+
+var TemplateWebSetupMagicLinkOTPHTML = template.RegisterHTML(
+	"web/setup_magic_link_otp.html",
+	components...,
+)
+
+var SetupMagicLinkOTPEmailSchema = validation.NewSimpleSchema(`
+	{
+		"type": "object",
+		"properties": {
+			"x_email": { "type": "string" }
+		},
+		"required": ["x_email"]
+	}
+`)
+
+func ConfigureSetupMagicLinkOTPRoute(route httproute.Route) httproute.Route {
+	return route.
+		WithMethods("OPTIONS", "POST", "GET").
+		WithPathPattern("/flows/setup_magic_link_otp")
+}
+
+type SetupMagicLinkOTPNode interface {
+}
+
+type SetupMagicLinkOTPHandler struct {
+	ControllerFactory ControllerFactory
+	BaseViewModel     *viewmodels.BaseViewModeler
+	Renderer          Renderer
+}
+
+func (h *SetupMagicLinkOTPHandler) GetData(r *http.Request, rw http.ResponseWriter, session *webapp.Session, graph *interaction.Graph) (map[string]interface{}, error) {
+	data := map[string]interface{}{}
+	baseViewModel := h.BaseViewModel.ViewModel(r, rw)
+
+	viewmodels.Embed(data, baseViewModel)
```

Comment:
We should show the alternatives list when setting up magic link, like setting up SMS 2fa. (signup only, not adding magic link in the settings page)

<img width="472" alt="Screenshot 2023-01-16 at 17 56 02" src="https://user-images.githubusercontent.com/388892/212741917-5527fdf6-f14d-4dac-8fe4-22e3331fdc71.png">

<img width="483" alt="Screenshot 2023-01-16 at 17 56 09" src="https://user-images.githubusercontent.com/388892/212741940-cc3aebf9-96a2-4239-a7a6-decd2af03fc5.png">

Reply:
Thanks updated 🙏 

![Screen Shot 2023-01-17 at 11 32 05](https://user-images.githubusercontent.com/3119725/212804029-c08269e0-fe8f-4953-81ee-d7e8983edd94.png)